src/locker.rs

```diff
@@ -18,12 +17,7 @@ extern "C" {
 }
 
 #[repr(C)]
-pub struct Locker<'a> {
-  has_lock: bool,
-  top_level: bool,
-  isolate: &'a mut Isolate,
-  phantom: PhantomData<&'a Isolate>,
-}
+pub struct Locker<'sc>([usize; 2], PhantomData<&'sc mut ()>);
```

Comment:
This struct layout is not verified anywhere and may cause problems. It's better to use an opaque buffer like this, at least we can assert it's the right size.